seisbench/generate/labeling.py

```diff
@@ -239,9 +274,22 @@ def label(self, X, metadata):
                 # Handle multi-window case
                 for j in range(X.shape[sample_dim]):
                     onset = metadata[label_column][j]
-                    label_val = gaussian_pick(
-                        onset=onset, length=X.shape[width_dim], sigma=self.sigma
-                    )
+                    if self.form == "gaussian":
+                        label_val = gaussian_pick(
+                            onset=onset, length=X.shape[width_dim], sigma=self.sigma
+                        )
+                    elif self.form == "triangular":
+                        label_val = triangular_pick(
+                            onset=onset, length=X.shape[width_dim], sigma=self.sigma
+                        )
+                    elif self.form == "box":
+                        label_val = box_pick(
+                            onset=onset, length=X.shape[width_dim], sigma=self.sigma
+                        )
+                    else:
+                        raise NotImplementedError(
+                            "Laber of form %s not implemented." % self.form
```

Comment:
As above (```ValueError``` > ```NotImplementedError```).

Reply:
Sounds like a good idea to condense the code, but I suggest we just check if `self.form` is a key of `self._labelform_fn_mapper`. With the code above, an error would be raised for 'triangular' and 'box' since they are 2nd/3rd to be checked.

See the latest committed code.